introspection-engine/connectors/sql-introspection-connector/src/introspection.rs

```diff
@@ -222,59 +195,36 @@ fn sql_enum_to_dml_enum(
     dml_enum
 }
 
-fn introspect_models(datamodel: &mut Datamodel, ctx: &Context) {
-    // collect m2m table names
-    let m2m_tables: Vec<String> = ctx
-        .schema
-        .table_walkers()
-        .filter(|table| is_prisma_1_point_1_or_2_join_table(*table) || is_prisma_1_point_0_join_table(*table))
-        .map(|table| table.name()[1..].to_string())
-        .collect();
+fn introspect_models(datamodel: &mut Datamodel, ctx: &mut Context<'_>) {
```

Comment:
This function should get its own module. I didn't do that to avoid a larger diff, but it is coming.